### PR TITLE
publishServoMoveTo with sane params and return

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 1]
```diff
@@ -601,7 +602,8 @@ public Double moveTo(Double newPos) {
       return newPos;
     }
 
-    boolean validMoveRequest = processMove(newPos, false, null);
+    broadcast("publishServoMoveTo", newPos);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: without the name of the servo publishing this, this is a pretty useless broadcast, no?

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 2]
```diff
@@ -713,32 +715,32 @@ public String publishServoDisable(ServoControl sc) {
...
+    log.info("publishServoEnable(servo)");
     return publishServoEnable((ServoControl) sc);
   }
 
   @Override
-  public ServoControl publishServoMoveTo(ServoControl sc) {
-    log.warn("publishServoMoveTo(servo)");
-    return sc;
+  public ServoMove publishServoMoveTo(Double pos) {
+    log.info("publishServoMoveTo({})");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: missing the pos if you are going to log this message here.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 3]
```diff
@@ -713,32 +715,32 @@ public String publishServoDisable(ServoControl sc) {
...
+    return new ServoMove(getName(), pos);
   }
 
   @Override
   public ServoSpeed publishServoSetSpeed(ServoControl sc) {
-    log.warn("publishServoSetSpeed(servo)");
+    log.info("publishServoSetSpeed(servo)");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: should probably log  servo.getName()   and subsitute it via {} in the log message.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 4]
```diff
@@ -713,32 +715,32 @@ public String publishServoDisable(ServoControl sc) {
...
     return new ServoSpeed(sc.getName(), sc.getSpeed());
   }
 
   @Override
   public ServoControl publishServoStop(ServoControl sc) {
-    log.warn("publishServoStop(servo)");
+    log.info("publishServoStop(servo)");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: should probably log  servo.getName()   and subsitute it via {} in the log message.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 5]
```diff
@@ -988,7 +990,7 @@ public void writeMicroseconds(int uS) {
    */
   @Override
   public ServoEvent publishServoStarted(String name, Double position) {
-    log.warn("publishServoStarted(name)");
+    log.info("publishServoStarted(name)");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: incorrect syntax for logging the name of the servo here.

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java [hunk 6]
```diff
@@ -1055,7 +1057,7 @@ public void startService() {
 
   @Override
   public String publishServoEnable(String name) {
-    log.warn("publishServoEnable(name)");
+    log.info("publishServoEnable(name)");
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: log the actual name here.. not just the string "name"

File: src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/abstracts/AbstractServo.java]: Still useful to the UI which inherently knows which servo ... but I totally agree - it should return a tuple that has name and position ... (future pr)

File: src/main/java/org/myrobotlab/service/interfaces/ServoControlPublisher.java [hunk 7]
```diff
@@ -9,7 +10,7 @@
  */
 public interface ServoControlPublisher {
 
-  public ServoControl publishServoMoveTo(ServoControl sc);
+  public ServoMove publishServoMoveTo(Double pos);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/interfaces/ServoControlPublisher.java]: This should take a ServoMove object, not a double, right?

File: src/main/java/org/myrobotlab/service/Servo.java [hunk 8]
```diff
@@ -128,6 +128,9 @@ protected boolean processMove(Double newPos, boolean blocking, Long timeoutMs) {
       log.info("{} is currently blocking - ignoring request to moveTo({})", getName(), newPos);
       return false;
     }
+    
+    broadcast("publishServoMoveTo", getTargetOutput());
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Servo.java]: This should be passing a servoMove object,... and why is this the output ? shouldn't it be publishing the input ?

Comment (supertick) [reply src/main/java/org/myrobotlab/service/Servo.java]: current method signature in AbstractServo
```java
  @Override
  public ServoMove publishServoMoveTo(Double pos) {
    log.debug("{}.publishServoMoveTo({})", getName(), pos);
    return new ServoMove(getName(), pos);
  }
```
could make it symmetrical, but not all interfaces adhere to that ... :shrug: - don't have much opinion

To answer the more significant question I think,
is this input or output - This method was made to be consumed by the ServoController ...
And it always has been output.  ie.  the ServoController.onServoMoveTo requires output - but potentially i can find this itself by using the service name to get the service to get the Servo.getTargetOutput()

... its a good question :shrug:

File: src/main/java/org/myrobotlab/service/interfaces/ServoControlPublisher.java (hunk 7 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/interfaces/ServoControlPublisher.java]: yeah ... it could

**Pull request merged**